server/api/batch_tallies.py

```diff
@@ -0,0 +1,184 @@
+from datetime import datetime
+import uuid
+from sqlalchemy.orm.session import Session
+from flask import request, jsonify, Request
+from werkzeug.exceptions import BadRequest, NotFound, Conflict
+
+from . import api
+from ..database import db_session
+from ..models import *  # pylint: disable=wildcard-import
+from ..auth import with_jurisdiction_access, with_election_access
+from ..util.process_file import (
+    process_file,
+    serialize_file,
+    serialize_file_processing,
+    UserError,
+)
+from ..util.csv_download import csv_response
+from ..util.csv_parse import decode_csv_file, parse_csv, CSVValueType, CSVColumnType
+
+BATCH_NAME = "Batch Name"
+
+
+def process_batch_tallies_file(
+    session: Session, jurisdiction: Jurisdiction, file: File
+):
+    def process():
```

Comment:
Not a huge deal, but I wonder if these internal function definitions are less efficient because they get redefined on every run of the outer function. Not a blocker here, just a thing to lookg into at some point.

Reply:
https://stackoverflow.com/questions/7839632/is-there-an-overhead-when-nesting-functions-in-python

this answer in particular was compelling: https://stackoverflow.com/a/50164012

> we can see that it [defining an inner function] takes less time than creating an empty dict (the fast way), so if you're doing anything non-trivial, this probably does not matter at all.

Reply:
great! today I learned.